opensrp-app/src/main/java/org/smartregister/login/task/RemoteLoginTask.java

```diff
@@ -162,6 +169,37 @@ protected LoginResponse doInBackground(Void... params) {
         return loginResponse;
     }
 
+    protected void fetchUserRole() {
+        Practitioner[] practitioners = getOpenSRPContext().httpAgent().fetchPractitioners();
+        PractitionerRole[] practitionerRoles = getOpenSRPContext().httpAgent().fetchPractitionerRoles();
+
+        if (practitioners == null) {
+            return;
+        }
+
+        Practitioner loggedInPractitioner = null;
+        for (Practitioner practitioner : practitioners) {
+            if (practitioner != null && mUsername.equals(practitioner.getUsername())) {
+                loggedInPractitioner = practitioner;
+            }
+        }
+
+        if (loggedInPractitioner != null && practitionerRoles != null) {
+
+            PractitionerRole loggedInPractitionerRole = null;
+            for (PractitionerRole practitionerRole : practitionerRoles) {
+                if (loggedInPractitioner.getIdentifier().equals(practitionerRole.getPractitionerIdentifier())) {
```

Comment:
Once this condition is satisfied, is it OK to break out of the loop for efficiency ?